cloudinit/config/schema.py

```diff
@@ -410,34 +420,53 @@ def _get_property_doc(schema: dict, prefix="    ") -> str:
     """Return restructured text describing the supported schema properties."""
     new_prefix = prefix + '    '
     properties = []
-    for prop_key, prop_config in schema.get('properties', {}).items():
-        # Define prop_name and description for SCHEMA_PROPERTY_TMPL
-        description = prop_config.get('description', '')
-
-        # Define prop_name and description for SCHEMA_PROPERTY_TMPL
-        properties.append(
-            SCHEMA_PROPERTY_TMPL.format(
-                prefix=prefix,
-                prop_name=prop_key,
-                description=_parse_description(description, prefix),
-                prop_type=_get_property_type(prop_config),
+    property_keys = [
+        schema.get("properties", {}),
+        schema.get("patternProperties", {}),
+    ]
+
+    for props in property_keys:
+        for prop_key, prop_config in props.items():
+            # Define prop_name and description for SCHEMA_PROPERTY_TMPL
+            description = prop_config.get("description", "")
+
+            # Define prop_name and description for SCHEMA_PROPERTY_TMPL
+            label = prop_config.get("label", prop_key)
+            properties.append(
+                SCHEMA_PROPERTY_TMPL.format(
+                    prefix=prefix,
+                    prop_name=label,
+                    description=_parse_description(description, prefix),
+                    prop_type=_get_property_type(prop_config),
+                )
             )
-        )
-        items = prop_config.get("items")
-        if items:
-            if isinstance(items, list):
-                for item in items:
+            items = prop_config.get("items")
+            if items:
+                if isinstance(items, list):
+                    for item in items:
+                        properties.append(
+                            _get_property_doc(item, prefix=new_prefix)
+                        )
+                elif isinstance(items, dict) and (
+                    items.get("properties") or items.get("patternProperties")
+                ):
                     properties.append(
-                        _get_property_doc(item, prefix=new_prefix))
-            elif isinstance(items, dict) and items.get('properties'):
-                properties.append(SCHEMA_LIST_ITEM_TMPL.format(
-                    prefix=new_prefix, prop_name=prop_key))
-                new_prefix += '    '
-                properties.append(_get_property_doc(items, prefix=new_prefix))
-        if 'properties' in prop_config:
-            properties.append(
-                _get_property_doc(prop_config, prefix=new_prefix))
-    return '\n\n'.join(properties)
+                        SCHEMA_LIST_ITEM_TMPL.format(
+                            prefix=new_prefix, prop_name=prop_key
```

Comment:
This needs label I think right?
```suggestion
                            prefix=new_prefix, prop_name=label
```